discovery/manager.go

```diff
@@ -190,9 +190,19 @@ func (m *Manager) updateGroup(poolKey poolKey, tgs []*targetgroup.Group) {
 			if _, ok := m.targets[poolKey]; !ok {
 				m.targets[poolKey] = make(map[string]*targetgroup.Group)
 			}
-			m.targets[poolKey][tg.Source] = tg
+
+			// No need to update when there aren't any actual changes.
+			// The k8s discoverer keeps sending updates for "kube-controller-manager" and
```

Comment:
We should drop this explanation about a specific target provider and just keep the first line. If we really want to keep the Kubernetes example we should mention why Kubernetes does this.